lib/internal/modules/esm/load.js

```diff
@@ -70,25 +70,30 @@ async function getSource(url, context) {
   return { __proto__: null, responseURL, source };
 }
 
+/**
+ * @param {URL} url URL to the module
+ * @param {ESModuleContext} context used to decorate error messages
+ * @returns {{ responseURL: string, source: string | BufferView }}
+ */
 function getSourceSync(url, context) {
-  const parsed = new URL(url);
-  const responseURL = url;
+  const { protocol, href } = url;
+  const responseURL = href;
   let source;
-  if (parsed.protocol === 'file:') {
-    source = readFileSync(parsed);
-  } else if (parsed.protocol === 'data:') {
-    const match = RegExpPrototypeExec(DATA_URL_PATTERN, parsed.pathname);
+  if (protocol === 'file:') {
+    source = readFileSync(url);
+  } else if (protocol === 'data:') {
+    const match = RegExpPrototypeExec(DATA_URL_PATTERN, url.pathname);
```

Comment:
worth pulling this off of `url` on line 79, in case there's a getter, so as to unconditionally trigger it?

Reply:
+1

Reply:
That seems wasteful, I'm not sure I understand what would be the upside of doing so

Reply:
Example benefits:
 - a getter that throws will do so at the top of the method, before expensive stuff happens
 - algorithm changes won't be as observable - it would otherwise be part of the API of this to only do a Get of "pathname" in this branch, and to only do one total, so refactors could become unintentional breaking changes
 - it becomes much clearer what properties are accessed since they're all accessed once, statically, at the top of the method

Reply:
If we really care about this, I think we should rather adapt the regex to take the `href` rather than the `pathname`.

> * a getter that throws will do so at the top of the method, before expensive stuff happens

It's still the first operation after the destructuring in that function.

> * algorithm changes won't be as observable - it would otherwise be part of the API of this to only do a Get of "pathname" in this branch, and to only do one total, so refactors could become unintentional breaking changes

You mean if we start needing `pathname` for other protocols? But a case could be made for the other properties of URL, what if we end up needing `search` or `origin`, should we get them as well just in case?

Reply:
yep